File: src/clients/FSharpGeneric/dtos.fs
(* Options:
Date: 2019-11-12 02:10:03
Version: 5.71
Tip: To override a DTO option, remove "//" prefix before updating
BaseUrl: https://localhost:5001

//GlobalNamespace: 
//MakeDataContractsExtensible: False
//AddReturnMarker: True
//AddDescriptionAsComments: True
//AddDataContractAttributes: False
//AddIndexesToDataMembers: False
//AddGeneratedCodeAttributes: False
//AddResponseStatus: False
//AddImplicitVersion: 
//ExportValueTypes: False
//IncludeTypes: 
//ExcludeTypes: 
//InitializeCollections: True
//AddNamespaces: 
*)

namespace TodoWorld.ServiceModel

open System
open System.Collections
open System.Collections.Generic
open System.Runtime.Serialization
open ServiceStack
open ServiceStack.DataAnnotations

    [<DataContract>]
    [<AllowNullLiteral>]
    type Todo() = 
        [<DataMember(Order=1)>]
        member val Id:Int64 = new Int64() with get,set

        [<DataMember(Order=2)>]
        member val Title:String = null with get,set

        [<DataMember(Order=3)>]
        member val Order:Int32 = new Int32() with get,set

        [<DataMember(Order=4)>]
        member val Completed:Boolean = new Boolean() with get,set

    [<DataContract>]
    [<AllowNullLiteral>]
    type HelloResponse() = 
        [<DataMember(Order=1)>]
        member val Result:String = null with get,set

        [<DataMember(Order=2)>]
        member val ResponseStatus:ResponseStatus = null with get,set

    [<DataContract>]
    [<AllowNullLiteral>]
    type GetTodoResponse() = 
        [<DataMember(Order=1)>]
        member val Result:Todo = null with get,set

        [<DataMember(Order=2)>]
        member val ResponseStatus:ResponseStatus = null with get,set

    [<DataContract>]
    [<AllowNullLiteral>]
    type GetTodosResponse() = 
        [<DataMember(Order=1)>]
        member val Results:List<Todo> = new List<Todo>() with get,set

        [<DataMember(Order=2)>]
        member val ResponseStatus:ResponseStatus = null with get,set

    [<DataContract>]
    [<AllowNullLiteral>]
    type CreateTodoResponse() = 
        [<DataMember(Order=1)>]
        member val Result:Todo = null with get,set

        [<DataMember(Order=2)>]
        member val ResponseStatus:ResponseStatus = null with get,set

    [<Route("/hello")>]
    [<Route("/hello/{Name}")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type Hello() = 
        interface IReturn<HelloResponse>
        [<DataMember(Order=1)>]
        member val Name:String = null with get,set

    [<Route("/hello-secure")>]
    [<Route("/hello-secure/{Name}")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type HelloSecure() = 
        interface IReturn<HelloResponse>
        [<DataMember(Order=1)>]
        member val Name:String = null with get,set

    [<Route("/todos/{Id}", "GET")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type GetTodo() = 
        interface IReturn<GetTodoResponse>
        [<DataMember(Order=1)>]
        member val Id:Int64 = new Int64() with get,set

    [<Route("/todos", "GET")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type GetTodos() = 
        interface IReturn<GetTodosResponse>

    [<Route("/todos", "POST")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type CreateTodo() = 
        interface IReturn<CreateTodoResponse>
        [<DataMember(Order=1)>]
        member val Title:String = null with get,set

        [<DataMember(Order=2)>]
        member val Order:Int32 = new Int32() with get,set

    [<Route("/todos/{Id}", "PUT")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type UpdateTodo() = 
        interface IReturnVoid
        [<DataMember(Order=1)>]
        member val Id:Int64 = new Int64() with get,set

        [<DataMember(Order=2)>]
        member val Title:String = null with get,set

        [<DataMember(Order=3)>]
        member val Order:Int32 = new Int32() with get,set

        [<DataMember(Order=4)>]
        member val Completed:Boolean = new Boolean() with get,set

    [<Route("/todos/{Id}", "DELETE")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type DeleteTodo() = 
        interface IReturnVoid
        [<DataMember(Order=1)>]
        member val Id:Int64 = new Int64() with get,set

    [<Route("/todos", "DELETE")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type DeleteTodos() = 
        interface IReturnVoid
        [<DataMember(Order=1)>]
        member val Ids:List<Int64> = new List<Int64>() with get,set

    [<Route("/todos/reset", "POST")>]
    [<DataContract>]
    [<AllowNullLiteral>]
    type ResetTodos() = 
        interface IReturnVoid

File: src/clients/VbGeneric/dtos.vb
' Options:
'Date: 2019-11-12 03:46:08
'Version: 5.71
'Tip: To override a DTO option, remove "''" prefix before updating
'BaseUrl: https://localhost:5001
'
'''GlobalNamespace: 
'''MakePartial: True
'''MakeVirtual: True
'''MakeDataContractsExtensible: False
'''AddReturnMarker: True
'''AddDescriptionAsComments: True
'''AddDataContractAttributes: False
'''AddIndexesToDataMembers: False
'''AddGeneratedCodeAttributes: False
'''AddResponseStatus: False
'''AddImplicitVersion: 
'''InitializeCollections: True
'''ExportValueTypes: False
'''IncludeTypes: 
'''ExcludeTypes: 
'''AddNamespaces: 
'''AddDefaultXmlNamespace: http://schemas.servicestack.net/types

Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Runtime.Serialization
Imports ServiceStack
Imports ServiceStack.DataAnnotations
Imports TodoWorld.ServiceModel.Types
Imports TodoWorld.ServiceModel

Namespace Global

    Namespace TodoWorld.ServiceModel

        <Route("/todos", "POST")>
        <DataContract>
        Public Partial Class CreateTodo
            Implements IReturn(Of CreateTodoResponse)
            <DataMember(Order:=1)>
            Public Overridable Property Title As String

            <DataMember(Order:=2)>
            Public Overridable Property Order As Integer
        End Class

        <DataContract>
        Public Partial Class CreateTodoResponse
            <DataMember(Order:=1)>
            Public Overridable Property Result As Todo

            <DataMember(Order:=2)>
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        <Route("/todos/{Id}", "DELETE")>
        <DataContract>
        Public Partial Class DeleteTodo
            Implements IReturnVoid
            <DataMember(Order:=1)>
            Public Overridable Property Id As Long
        End Class

        <Route("/todos", "DELETE")>
        <DataContract>
        Public Partial Class DeleteTodos
            Implements IReturnVoid
            Public Sub New()
                Ids = New List(Of Long)
            End Sub

            <DataMember(Order:=1)>
            Public Overridable Property Ids As List(Of Long)
        End Class

        <Route("/todos/{Id}", "GET")>
        <DataContract>
        Public Partial Class GetTodo
            Implements IReturn(Of GetTodoResponse)
            <DataMember(Order:=1)>
            Public Overridable Property Id As Long
        End Class

        <DataContract>
        Public Partial Class GetTodoResponse
            <DataMember(Order:=1)>
            Public Overridable Property Result As Todo

            <DataMember(Order:=2)>
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        <Route("/todos", "GET")>
        <DataContract>
        Public Partial Class GetTodos
            Implements IReturn(Of GetTodosResponse)
        End Class

        <DataContract>
        Public Partial Class GetTodosResponse
            Public Sub New()
                Results = New List(Of Todo)
            End Sub

            <DataMember(Order:=1)>
            Public Overridable Property Results As List(Of Todo)

            <DataMember(Order:=2)>
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        <Route("/hello")>
        <Route("/hello/{Name}")>
        <DataContract>
        Public Partial Class Hello
            Implements IReturn(Of HelloResponse)
            <DataMember(Order:=1)>
            Public Overridable Property Name As String
        End Class

        <DataContract>
        Public Partial Class HelloResponse
            <DataMember(Order:=1)>
            Public Overridable Property Result As String

            <DataMember(Order:=2)>
            Public Overridable Property ResponseStatus As ResponseStatus
        End Class

        <Route("/hello-secure")>
        <Route("/hello-secure/{Name}")>
        <DataContract>
        Public Partial Class HelloSecure
            Implements IReturn(Of HelloResponse)
            <DataMember(Order:=1)>
            Public Overridable Property Name As String
        End Class

        <Route("/todos/reset", "POST")>
        <DataContract>
        Public Partial Class ResetTodos
            Implements IReturnVoid
        End Class

        <Route("/todos/{Id}", "PUT")>
        <DataContract>
        Public Partial Class UpdateTodo
            Implements IReturnVoid
            <DataMember(Order:=1)>
            Public Overridable Property Id As Long

            <DataMember(Order:=2)>
            Public Overridable Property Title As String

            <DataMember(Order:=3)>
            Public Overridable Property Order As Integer

            <DataMember(Order:=4)>
            Public Overridable Property Completed As Boolean
        End Class
    End Namespace

    Namespace TodoWorld.ServiceModel.Types

        <DataContract>
        Public Partial Class Todo
            <DataMember(Order:=1)>
            Public Overridable Property Id As Long

            <DataMember(Order:=2)>
            Public Overridable Property Title As String

            <DataMember(Order:=3)>
            Public Overridable Property Order As Integer

            <DataMember(Order:=4)>
            Public Overridable Property Completed As Boolean
        End Class
    End Namespace
End Namespace

